PowerShell/JumpCloud Module/Public/Systems/Get-JCSystemApp.ps1

```diff
@@ -43,298 +39,289 @@ function Get-JCSystemApp () {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                } elseif ($SystemOS) {
+                                    Get-JcSdkSystemInsightLinuxPackage -Filter @("name:eq:$SoftwareName") | ForEach-Object {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                }
+                            } elseif ($SystemID) {
+                                if ($SoftwareVersion) {
+                                    Write-Error "Cannot search for software version on Linux without software name."
+                                } else {
+                                    Get-JcSdkSystemInsightLinuxPackage -Filter @("system_id:eq:$SystemID") | ForEach-Object {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                }
+
+                            } elseif ($SystemOS) {
+                                if ($SoftwareVersion) {
+                                    Write-Error "Cannot search for software version on Linux without software name."
+                                } else {
+                                    Get-JcSdkSystemInsightLinuxPackage | ForEach-Object {
+                                        [void]$resultsArrayList.Add($_)
+                                    }
+                                }
                             }
-                            Write-Debug $URL
                         }
 
                     }
-                    if ($Parallel) {
-                        $resultsArrayList = Get-JCResults -URL $URL -Method "GET" -limit $limit -parallel $true
-                    } else {
-                        $resultsArrayList = Get-JCResults -URL $URL -Method "GET" -limit $limit
-                    }
                 } elseif ($SoftwareName) {
-                    $SoftwareName = [System.Web.HttpUtility]::UrlEncode($SoftwareName)
-                    # Search each apps endpoint for software name
-                    foreach ($os in @('MacOs', 'Windows', 'Linux')) {
-                        if ($os -eq 'MacOs') {
+                    # Loop through each OS and get the results
+                    Write-Debug "SoftwareName only passed. Getting all software with name $SoftwareName"
+                    foreach ($os in @('Windows', 'MacOs', 'Linux')) {
+                        if ($os -eq 'Windows') {
+                            Get-JcSdkSystemInsightProgram -Filter @("name:eq:$SoftwareName") | ForEach-Object {
+                                [void]$resultsArrayList.Add($_)
+                            }
+                        } elseif ($os -eq 'MacOs') {
+                            $macOsSoftwareName = $SoftwareName
```

Comment:
Could we condense this using .ToLower() or are the SoftwareNames case sensitive? It could reduce the amount of conditions